zlib-rs/src/allocate.rs

```diff
@@ -160,6 +160,14 @@ impl Allocator<'static> {
     };
 }
 
+fn null_is_none<T>(ptr: *mut T) -> Option<*mut T> {
```

Comment:
Would it make sense to return `Option<NonNull<T>>` instead? Then it would just be a `NonNull::new()` to get the `Option` and the return value is half as big.